### PR TITLE
🌐 Absolutify attachment URLs during flow migration

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: legacy/definition.go [hunk 1]
```diff
@@ -440,6 +440,17 @@ func migrateAction(baseLanguage utils.Language, a Action, localization flows.Loc
 			quickReplies = TransformTranslations(legacyQuickReplies)
 		}
 
+		for lang, attachment := range media {
+			parts := strings.SplitN(attachment, ":", 2)
+			mediaType := parts[0]
+			mediaURL := parts[1]
+
+			if strings.Contains(mediaType, "/") {
```
Comment (ericnewcomer) [inline legacy/definition.go]: Shouldn't we be looking for `@` ?

Comment (rowanseymour) [reply legacy/definition.go]: this is differentiating between _foo/bar:relativeurl_ and _foo:template_ .. the thing about the latter is it doesn't necessarily contain a `@` expression. This matches how `ReplyAction` determines whether or not to add `STORAGE_URL` in the legacy engine.

File: legacy/definition.go [hunk 2]
```diff
@@ -440,6 +440,17 @@ func migrateAction(baseLanguage utils.Language, a Action, localization flows.Loc
 			quickReplies = TransformTranslations(legacyQuickReplies)
 		}
 
+		for lang, attachment := range media {
+			parts := strings.SplitN(attachment, ":", 2)
+			mediaType := parts[0]
+			mediaURL := parts[1]
```
Comment (ericnewcomer) [inline legacy/definition.go]: Should we enforce that it doesn't (or alternatively does) end in a `/` so we don't double slash below?

File: legacy/definition.go (hunk 1 above)
Comment (ericnewcomer) [reply legacy/definition.go]: Ah, makes sense

File: legacy/definition.go (hunk 2 above)
Comment (rowanseymour) [reply legacy/definition.go]: The legacy engine assumes they don't start with `/` but I'll add this for good measure

**Pull request merged**